### ISSUE TITLE
Update S3237: Rule should handle expression body accessors

### ISSUE THREAD

**Issue by Evangelink** [SonarSource/sonar-csharp#1021]
### Description

The rule currently won't report on properties using the new syntax (expression body accessor).

### Repro steps

```
namespace ConsoleApp8
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Hello, world!");
        }

        public Program(string id)
        {
            if (string.IsNullOrEmpty(id)) throw new ArgumentNullException(nameof(id));
        }

        private string id;

        public string Id
        {
            get => null;
            set => id = 10; // Rule won't raise if the value is not used here
        }
    }
}
```

### Expected behavior

The issue should be raised.

### Actual behavior

The issue is not raised.
